File: le_nhut_cuong_chieut5/lib/models/reminder.dart
class Reminder {
  final String id;
  final DateTime remindAt;
  final String method;
  final String? eventId; // 👈 thêm

  Reminder({
    required this.id,
    required this.remindAt,
    required this.method,
    this.eventId,
  });

  factory Reminder.fromJson(Map<String, dynamic> json) {
    return Reminder(
      id: json['_id'],
      remindAt: DateTime.parse(json['remindAt']),
      method: json['method'] ?? 'notification',
      eventId: json['event'] is Map
          ? json['event']['_id']
          : json['event'],
    );
  }
}

File: backend/models/Reminder.js
import mongoose from "mongoose";

const reminderSchema = new mongoose.Schema(
  {
    event: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Event",
      required: true,
    },

    remindAt: { type: Date, required: true },

    method: {
      type: String,
      enum: ["notification", "email"],
      default: "notification",
    },

    owner: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
      required: true,
    },
  },
  { timestamps: true }
);

export default mongoose.model("Reminder", reminderSchema);
